BridgeAppSDK/SBAUserWrapper+Bridge.swift

```diff
@@ -0,0 +1,311 @@
+        
+        // If the user is not logged in or consented then do not attempt login
+        // Just return with an error
+        guard self.loginVerified && self.consentVerified else {
+            let code = !self.loginVerified ? SBBErrorCode.NoCredentialsAvailable.rawValue : SBBErrorCode.ServerPreconditionNotMet.rawValue
+            let error = NSError(domain: SBAUserErrorDomain, code: code, userInfo: nil)
+            self.callCompletionOnMain(error, completion: completion)
+            return
+        }
+        
+        // Make sure that 
+        SBAUserBridgeManager.ensureSignedInWithCompletion { [weak self] (_, error) in
+            guard (self != nil) else { return }
+            
+            if (error != nil) && error!.code == SBBErrorCode.ServerPreconditionNotMet.rawValue {
+                // If the server returns a 412 after login and consent have been verified then need to reconsent
+                self!.consentVerified = false
+                self!.consentSignature = nil
+            }
+            
+            self!.callCompletionOnMain(error, completion: completion)
+        }
+    }
+    
+    private func signInUser(username: String, password: String, completion: ((NSError?) -> Void)?) {
+        
+        SBAUserBridgeManager.signIn(username, password: password) { [weak self] (responseObject, error) in
+            guard (self != nil) else { return }
+            
+            // If there was an error and it is *not* the consent error then call completion and exit
+            let requiresConsent = (error != nil) && error!.code == SBBErrorCode.ServerPreconditionNotMet.rawValue
+            guard ((error != nil) && !requiresConsent) else {
+                self!.callCompletionOnMain(error, completion: completion)
+                return
+            }
+            
+            // If signed in successfully, then set the registered and verified flags to true
+            self!.hasRegistered = true
+            self!.loginVerified = true
+            
+            // Copy info from the user session response object
+            if let response = responseObject as? SBAUserSessionInfoWrapper {
+                self!.copyFromUserSession(response)
+            }
+            
+            if let consentSignature = self!.consentSignature where requiresConsent {
+                // If there is a consent signature object stored for this user then attempt
+                // sending consent once signed in.
+                self!.sendUserConsented(consentSignature, completion: completion)
+            }
+            else {
+                // otherwise, we are done. Set the flag that the consent has been verified and 
+                // call the completion
+                self!.consentVerified = !requiresConsent
+                self!.callCompletionOnMain(error, completion: completion)
+            }
+        }
+    }
+    
+    private func copyFromUserSession(response: SBAUserSessionInfoWrapper) {
```

Comment:
probably I'm just conditioned from my pre-ARC days using Core Foundation, but I feel like this should be named "updateFrom..." or some such rather than "copyFrom...".